File: csharp/UrakawaToolkit/TextMedia.cs
using System;

namespace urakawa.media
{
	/// <summary>
	/// TextMedia represents a text string
	/// </summary>
	public class TextMedia : ITextMedia
	{
		private string mTextString;

		
		//internal constructor encourages use of MediaFactory to create TextMedia objects
		internal TextMedia()
		{
		}

		/// <summary>
		/// this override is useful while debugging
		/// </summary>
		/// <returns></returns>
		public override string ToString()
		{
			return "TextMedia";
		}
		#region ITextMedia Members

		public string getText()
		{
			return mTextString;
		}

		public void setText(string text)
		{
			if (text == null)
			{
				throw new exception.MethodParameterIsNullException("TextMedia.setText(null) caused MethodParameterIsNullException");
				return;
			}

			if (text.Length == 0)
			{
				throw new exception.MethodParameterIsEmptyStringException("TextMedia.setText(" + 
					text + ") caused MethodParameterIsEmptyStringException");

				//causing a return here might be too oppositional, what if you are using an empty string?
				//(assuming it even matters in c#)
			}
			
			mTextString = text;
		}

		#endregion

		#region IMedia Members

		public bool isContinuous()
		{
			return false;
		}

		public bool isDiscrete()
		{
			return true;
		}

		public urakawa.media.MediaType getType()
		{
			return MediaType.TEXT;
		}

		#endregion
	}
}

File: branches/architecture/it2-experiments/src/org/daisy/urakawa/media/TextMedia.java
package org.daisy.urakawa.media;

import org.daisy.urakawa.exceptions.MethodParameterIsEmptyStringException;
import org.daisy.urakawa.exceptions.MethodParameterIsNullException;

/**
 * Just text, no structure (e.g. HTML formating), no time (e.g. SMIL).
 * {@link Media#isContinuous()} should return false for static text fragments like this.
 * {@link Media#getType()} should return MediaType.TEXT
 */
public interface TextMedia extends Media {
    /**
     * @param text Cannot be NULL or empty String.
     * @tagvalue Exceptions "MethodParameterIsNull, MethodParameterIsEmptyString"
     */
    public void setText(String text) throws MethodParameterIsNullException, MethodParameterIsEmptyStringException;

    /**
     * @return the text. Cannot be NULL or empty String.
     */
    public String getText();
}
